src/include/distributed/citus_safe_lib.h

```diff
@@ -0,0 +1,27 @@
+/*-------------------------------------------------------------------------
+ *
+ * safe_lib.h
+ *
+ * This file contains helper macros to make the _s suffix string and buffer
+ * functions from safestringlib easier to use.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef CITUS_safe_lib_H
+#define CITUS_safe_lib_H
+
+#include "safe_lib.h"
+
+extern void ErrnoToEreport(char *functionName, errno_t err);
+
+#define s_func_to_ereport(functionName, ...) \
+	ErrnoToEreport(# functionName, (functionName) (__VA_ARGS__))
+
+#define strncpy_ereport(...) s_func_to_ereport(strncpy_s, __VA_ARGS__)
+#define strcpy_ereport(...) s_func_to_ereport(strcpy_s, __VA_ARGS__)
+#define memcpy_ereport(...) s_func_to_ereport(memcpy_s, __VA_ARGS__)
```

Comment:
Naming is a little obscure. Wonder if we should maybe even break clunky libc naming and ca;; it `SafeMemoryCopy`.

Reply:
I renamed them to SafeMemcpy etc. I think it's nice to not chaneg the libc naming to make it clear that they match with them.